python/sglang/srt/layers/moe/topk.py

```diff
@@ -122,9 +124,20 @@ def grouped_topk(
     )  # [n, e]
     tmp_scores = scores.masked_fill(~score_mask.bool(), 0.0)  # [n, e]
     topk_weights, topk_ids = torch.topk(tmp_scores, k=topk, dim=-1, sorted=False)
+    if share_fusion:
+        topk_ids[:, -1] = torch.randint(low=num_experts,
+                                        high=num_experts + share_fusion,
+                                        size=(topk_ids.size(0), ),
+                                        dtype=topk_ids.dtype,
+                                        device=topk_ids.device)
+        topk_weights[:, -1] = topk_weights[:, :-1].sum(dim=-1) * 1.0 / 2.5
```

Comment:
nit: wondering whether `a*1.0` will be optimized as `a`, or will pytorch actually call kernels to do a computation. If the latter, maybe we can remove `*1.0` to speed up a tiny little bit